// react-floating-button-menu/es/FloatingMenu.js
var _templateObject = _taggedTemplateLiteralLoose(['\n  display: flex;\n  box-sizing: border-box;\n  padding-left: 0;\n  \n  *, *:before, *:after {\n    box-sizing: inherit;\n  }\n'], ['\n  display: flex;\n  box-sizing: border-box;\n  padding-left: 0;\n  \n  *, *:before, *:after {\n    box-sizing: inherit;\n  }\n']),
    _templateObject2 = _taggedTemplateLiteralLoose(['\n   list-style: none;\n   margin: 0;\n   padding: 0;\n'], ['\n   list-style: none;\n   margin: 0;\n   padding: 0;\n']);

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _taggedTemplateLiteralLoose(strings, raw) { strings.raw = raw; return strings; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

var Container = styled.div(_templateObject);

var StyledUl = styled.ul(_templateObject2);

var FloatingMenu = function (_Component) {
  _inherits(FloatingMenu, _Component);

  function FloatingMenu() {
    var _temp, _this, _ret;

    _classCallCheck(this, FloatingMenu);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.state = {
      isOpen: false
    }, _this.toggleMenu = function () {
      var _this2;

      return (_this2 = _this).__toggleMenu__REACT_HOT_LOADER__.apply(_this2, arguments);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  FloatingMenu.prototype.__toggleMenu__REACT_HOT_LOADER__ = function __toggleMenu__REACT_HOT_LOADER__(event) {
    event.preventDefault();
    this.setState({
      isOpen: !this.state.isOpen
    });
  };

  FloatingMenu.prototype.render = function render() {
    var _props = this.props,
        slideSpeed = _props.slideSpeed,
        direction = _props.direction,
        size = _props.size;
    var isOpen = this.state.isOpen;

    var childrenWithProps = React.Children.map(this.props.children, function (child) {
      return React.cloneElement(child, {
        isOpen: isOpen,
        slideSpeed: slideSpeed,
        direction: direction,
        size: size
      });
    });

    return React.createElement(
      Container,
      this.props,
      React.createElement(
        StyledUl,
        { onClick: this.toggleMenu },
        childrenWithProps
      )
    );
  };

  return FloatingMenu;
}(Component);

FloatingMenu.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  slideSpeed: PropTypes.number,
  direction: PropTypes.string,
  size: PropTypes.shape({
    main: PropTypes.number,
    child: PropTypes.number
  })
};
FloatingMenu.defaultProps = {
  background: 'palevioletred',
  slideSpeed: 500,
  direction: 'down',
  size: {
    main: 56,
    child: 56
  }
};
var _default = FloatingMenu;


export default _default;
;

var _temp2 = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(Container, 'Container', 'src/FloatingMenu.js');

  __REACT_HOT_LOADER__.register(StyledUl, 'StyledUl', 'src/FloatingMenu.js');

  __REACT_HOT_LOADER__.register(FloatingMenu, 'FloatingMenu', 'src/FloatingMenu.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/FloatingMenu.js');
}();

;
