// react-floating-button-menu/MainButton.js
'use strict';

exports.__esModule = true;

var _templateObject = _taggedTemplateLiteralLoose(['\n  background-color: ', ';\n  display: flex;\n  position: absolute;\n  border: none;\n  border-radius: 50%;\n  box-shadow: 0 0 4px rgba(0,0,0,.14),0 4px 8px rgba(0,0,0,.28);\n  cursor: pointer;\n  outline: none;\n  padding: 0;\n  -webkit-user-drag: none;\n  font-weight: bold;\n  color: #f1f1f1;\n  width: ', 'px;\n  height: ', 'px;\n  justify-content: center;\n  align-items: center;\n  -webkit-transition: -webkit-transform 300ms;\n  transition: transform 300ms;\n  ', '\n'], ['\n  background-color: ', ';\n  display: flex;\n  position: absolute;\n  border: none;\n  border-radius: 50%;\n  box-shadow: 0 0 4px rgba(0,0,0,.14),0 4px 8px rgba(0,0,0,.28);\n  cursor: pointer;\n  outline: none;\n  padding: 0;\n  -webkit-user-drag: none;\n  font-weight: bold;\n  color: #f1f1f1;\n  width: ', 'px;\n  height: ', 'px;\n  justify-content: center;\n  align-items: center;\n  -webkit-transition: -webkit-transform 300ms;\n  transition: transform 300ms;\n  ', '\n']),
    _templateObject2 = _taggedTemplateLiteralLoose(['\n    position: absolute;\n    fill: ', ';\n    opacity: ', ';\n  '], ['\n    position: absolute;\n    fill: ', ';\n    opacity: ', ';\n  ']),
    _templateObject3 = _taggedTemplateLiteralLoose(['\n    position: absolute;\n    fill: ', ';\n    opacity: ', '\n  '], ['\n    position: absolute;\n    fill: ', ';\n    opacity: ', '\n  ']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _taggedTemplateLiteralLoose(strings, raw) { strings.raw = raw; return strings; } /* eslint-disable react/no-unused-prop-types */


var Wrapper = _styledComponents2.default.a(_templateObject, function (props) {
  return props.backgroundColor;
}, function (props) {
  return props.size.main;
}, function (props) {
  return props.size.main;
}, function (prop) {
  return prop.isOpen ? '\n    -webkit-transform: rotate(180deg);\n    transform: rotate(180deg);\n  ' : '';
});

var IconResting = function IconResting(props) {
  return (0, _styledComponents2.default)(props.iconResting)(_templateObject2, props.iconColor, props.isOpen ? 0 : 1);
};

var IconActive = function IconActive(props) {
  return (0, _styledComponents2.default)(props.iconActive)(_templateObject3, props.iconColor, props.isOpen ? 1 : 0);
};

var MainButton = function (_Component) {
  _inherits(MainButton, _Component);

  function MainButton() {
    _classCallCheck(this, MainButton);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  MainButton.prototype.render = function render() {
    var StyledIconResting = IconResting(this.props);
    var StyledIconActive = IconActive(this.props);

    return _react2.default.createElement(
      Wrapper,
      this.props,
      _react2.default.createElement(StyledIconResting, null),
      _react2.default.createElement(StyledIconActive, null)
    );
  };

  return MainButton;
}(_react.Component);

MainButton.propTypes = {
  iconResting: _propTypes2.default.func.isRequired,
  iconActive: _propTypes2.default.func.isRequired,
  iconColor: _propTypes2.default.string,
  isOpen: _propTypes2.default.bool,
  size: _propTypes2.default.shape({
    main: _propTypes2.default.number,
    child: _propTypes2.default.number
  })
};
MainButton.defaultProps = {
  iconColor: '#ff1744',
  size: {
    main: 56,
    child: 56
  },
  isOpen: false
};
var _default = MainButton;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(Wrapper, 'Wrapper', 'src/MainButton.js');

  __REACT_HOT_LOADER__.register(IconResting, 'IconResting', 'src/MainButton.js');

  __REACT_HOT_LOADER__.register(IconActive, 'IconActive', 'src/MainButton.js');

  __REACT_HOT_LOADER__.register(MainButton, 'MainButton', 'src/MainButton.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/MainButton.js');
}();

;
